Vim does not use spack's ncurses

``` bash
$ spack install vim
==> Installing vim
==> ncurses is already installed in spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/ncurses-6.0-75hnzrecoib5eqdvfchl6n2jgcxc6dgo
==> Using cached archive: spack/var/spack/cache/vim/vim-8.0.0503.tar.gz
==> Staging archive: spack/var/spack/stage/vim-8.0.0503-55us7ijakx452ll3suegl6ocx7nzcmfi/v8.0.0503.tar.gz
==> Created stage in spack/var/spack/stage/vim-8.0.0503-55us7ijakx452ll3suegl6ocx7nzcmfi
==> Ran patch() for vim
==> Building vim [AutotoolsPackage]
==> Executing phase : 'autoreconf'
==> Executing phase : 'configure'
==> Error: ProcessError: Command exited with status 1:
    'spack/var/spack/stage/vim-8.0.0503-55us7ijakx452ll3suegl6ocx7nzcmfi/vim-8.0.0503/configure' '--prefix=spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/vim-8.0.0503-55us7ijakx452ll3suegl6ocx7nzcmfi' '--enable-fail-if-missing' '--with-features=normal' '--enable-pythoninterp=no' '--enable-rubyinterp=no' '--enable-luainterp=no' '--enable-perlinterp=no' '--enable-gui=no' '--without-x'
spack/lib/spack/spack/build_systems/autotools.py:266, in configure:
     258      def configure(self, spec, prefix):
     259          """Runs configure with the arguments specified in
     260          :py:meth:`~.AutotoolsPackage.configure_args`
     261          and an appropriately set prefix.
     262          """
     263          options = ['--prefix={0}'.format(prefix)] + self.configure_args()
     264
     265          with working_dir(self.build_directory, create=True):
  >> 266              inspect.getmodule(self).configure(*options)
```

spack-build.out
``` bash
...
checking --with-tlib argument... empty: automatic terminal library selection
checking for tgetent in -ltinfo... no
checking for tgetent in -lncurses... no
checking for tgetent in -ltermlib... no
checking for tgetent in -ltermcap... no
checking for tgetent in -lcurses... no
no terminal library found
checking for tgetent()... configure: error: NOT FOUND!
```

Vim does not build on Centos 7 unless ```ncurses-devel``` or equevalent package is installed because the vim's configure will search for libncurses, but not libncursesw. 

The fix is to explicitly point vim's configure to ncuresew using ```--with-tlib=ncursesw```

--- a/var/spack/repos/builtin/packages/vim/package.py
+++ b/var/spack/repos/builtin/packages/vim/package.py
@@ -90,6 +90,8 @@ class Vim(AutotoolsPackage):
 
         configure_args = ["--enable-fail-if-missing"]
 
+        configure_args.append("--with-tlib=ncursesw")
+
         configure_args.append("--with-features=" + feature_set)
 
         if '+python' in spec:

